workflow/controller/controller.go

```diff
@@ -502,7 +502,6 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	logCtx := log.WithFields(log.Fields{"key": key, "action": action})
 	logCtx.Info("cleaning up pod")
 	err := func() error {
-		pods := wfc.kubeclientset.CoreV1().Pods(namespace)
```

Comment:
This doesn't actually look up pods, it only invokes the API. The patch/delete calls are the actual calls to K8s. I don't see any benefits this brings.